### PR TITLE
integrating add broadcasts to backend

### PR DESCRIPTION
**Pull request by Aman16-ai**
## Issue that this pull request solves

 Closes: #776 

### Brief description of what is fixed or changed

- uploading add broadcasts form data to backend

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots
Screenshot (784)
Screenshot (785)

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx [hunk 1]
```diff
@@ -70,17 +84,54 @@ export function AddBroadcasts() {
...
+      const data = await response.json();
+      setIsUploadingData(false)
+      setToastMessage("broadcats added successfully!")
+      setOpenToast(true)
+      console.log(data)
+    }
+    catch (err) {
+      console.log(err)
+      setIsUploadingData(false)
+      setToastMessage("Something went wrong");
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:124]: please change the message to `something went wrong!`

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx [hunk 2]
```diff
@@ -93,113 +144,124 @@ export function AddBroadcasts() {
...
+              </div>
+              <input
+                type="text"
+                ref={imageRef}
+                onKeyDown={e => {
+                  if (e.key === "Enter") {
+                    addImageUrl();
+                  }
+                }}
+                placeholder="Enter Image url (Hit enter to add image url)"
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:215]: please change this message to `Enter Image url`

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/add-broadcasts.module.scss [hunk 3]
```diff
@@ -1,3 +1,10 @@
+.add-broadcasts-container {
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+}
 .editor {
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/add-broadcasts.module.scss:8]: please format this file.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@Aman16-ai The form validation for Add Broadcast seems to mis-behave. Can you please fix it? Also, it's not triggering the api once I click on submit button.
ref -
Screenshot 2022-04-06 at 4 26 19 PM

**Comment by Aman16-ai** [comment]
> @Aman16-ai The form validation for Add Broadcast seems to mis-behave. Can you please fix it? Also, it's not triggering the api once I click on submit button. ref - Screenshot 2022-04-06 at 4 26 19 PM

Yes I'll fix the form validation. Api is not triggering in your case because you didn't hit enter key in imageurl field to add url in imageurl array.

**Comment by Aman16-ai** [comment]
suneditor resets the previous state of the formdata object. That's why validation errors occur
This issue is also listed on suneditor github repo : https://github.com/mkhstar/suneditor-react/issues/267

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx [hunk 4]
```diff
@@ -70,17 +84,54 @@ export function AddBroadcasts() {
...
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${localStorage.getItem('token')}`
+        },
+        body: JSON.stringify(formData)
+      });
+      const data = await response.json();
+      setIsUploadingData(false)
+      setToastMessage("broadcats added successfully!")
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:117]: Even after api is giving `400` as response, I am seeing this toast message.

Also, please correct the message to `broadcast added successfully!`

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx [hunk 5]
```diff
@@ -70,17 +84,54 @@ export function AddBroadcasts() {
...
+      setOpenToast(true)
+      console.log(data)
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:119]: please remove this console statement

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx [hunk 6]
```diff
@@ -70,17 +84,54 @@ export function AddBroadcasts() {
...
+    }
+    catch (err) {
+      console.log(err)
+      setIsUploadingData(false)
+      setToastMessage("Something went wrong");
+      setOpenToast(true)
+      setSeverity("error")
+    }
+  }
   const onSubmit = e => {
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:129]: After successful submission of the form, the form value should reset.
If any error occurs, it shouldn't reset.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@Aman16-ai The data is not getting submitted in the backend and I am seeing  `400` bad request error with the toast message of `broadcast added successfully!` even after providing the correct entry. Please correct this along with the requested changes

File: frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx (hunk 6 above)
Comment (Aman16-ai) [reply frontend/src/pages/Admin/Components/Broadcast/AddBroadcasts/AddBroadcasts.jsx:150]: - fixed toasts status on error response
- Resetable form

https://user-images.githubusercontent.com/74807466/166094348-9ef0faef-0c53-4ecc-8dd7-df1443ccdee8.mp4

**Pull request closed without merging**